bootstrap

```diff
@@ -190,6 +190,10 @@ lazy_build() {
     # then we'll skip the worktree handling because main_worktree_dir will
     # be empty.
     main_worktree_dir=`git worktree list --porcelain 2>/dev/null|sed 's/^worktree //;q'`
+    # If gitdir isn't the default value, assume that we are in a submodule.
+    if [ "$gitdir" != ".git" ] ; then
+      main_worktree_dir=`echo $gitdir|sed 's/\/.git\/modules\//\//;q'`
+    fi
```

Comment:
I'm a bit unsure about this part.  What does `git worktree list --porcelain` report in your situation?

I worry that `$gitdir` can be something other than `.git` in other situations.  For example, if `$GIT_DIR` is set I get a full path on it:

```
olly@gemse:~/git/xapian$ GIT_DIR=/home/olly/git/xapian/.git git rev-parse --git-dir
/home/olly/git/xapian/.git
```

Perhaps it should be testing for the magic substring instead, which is easy to do with `case`:

```
case $gitdir in
*/.git/modules/*)
    main_worktree_dir=`...` ;;
esac
```

I guess this won't handle using submodules and worktrees together.  It'd be nicer if that worked, though if you don't use worktrees I guess that can be left for someone who wants both to take care of.

Also, I'm wondering why there's a `;q` in the `sed` expression here, since I'd have thought this would always be a single line (unless the path actually contains a newline in which case attempting the replace on all lines is more likely to do the right thing) - I'm suspecting it was just copied from the `sed` command just above, where we are explicitly wanting to cull all lines except the first - the output there looks like this:

```
$ git worktree list --porcelain 
worktree /home/olly/git/xapian
HEAD def7279ac6884346e746d063641f13b9fd5fbf35
branch refs/heads/master

worktree /home/olly/git/xapian-1.4
HEAD eea3d616a7371ca97f8615f0d226697fe2f94608
branch refs/heads/RELEASE/1.4
```

I'd also pick a different delimiter to avoid [leaning toothpick syndrome](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome), so something like: `sed 's|/.git/modules/|/|'`

Reply:
This I'll have to test some more.  Let me get back to you on that.

Sorry for not replying when these came through - I was traveling when the initial reviews happened and clearly scrolled past these when I got back!